app/static/js/profile.js

```diff
@@ -0,0 +1,244 @@
+        option.attr("selected", "selected");
+      }
+      options.push(option);
+    }
+
+    $("#summerYear").empty().append(options);
+  }
+
+  var today = new Date().toISOString().split('T')[0];
+  document.getElementById('date').value = today;
+  updateYearOptions();
+
+  document.getElementById('date').addEventListener('change', updateYearOptions);
+
+  $('#summerExperienceForm').on('submit', function(event) {
+    event.preventDefault(); 
+    var formData = new FormData(this); 
+    var actionUrl = $(this).attr('action'); 
+    
+    $.ajax({
+      url: actionUrl,
+      type: 'POST',
+      data: formData,
+      contentType: false,
+      processData: false,
+      success: function(response) {
+        $('#pills-training').html(response);
+        $('#trainingEvents').tab('show');
+      },
+      error: function(xhr, status, error) {
+        console.error('Error:', error);
+      }
+    });
+  });
+
+  document.getElementById('edit-proposal-button').addEventListener('click', function() {
+    toggleEditMode(true);
+  });
+
+  document.getElementById('cancel-edit-button').addEventListener('click', function() {
+    toggleEditMode(false);
+  });
+
+  function toggleEditMode(isEditMode) {
+    document.querySelectorAll('input, textarea, select').forEach(element => {
+      if (element.type === 'radio' || element.type === 'checkbox' || element.tagName === 'SELECT') {
+        element.disabled = !isEditMode;
+      } else {
+        element.readOnly = !isEditMode;
+      }
+    });
+    document.getElementById('edit-buttons').style.display = isEditMode ? 'block' : 'none';
+    document.getElementById('view-buttons').style.display = isEditMode ? 'none' : 'block';
+  }
+
+  toggleEditMode(false);
+
+  var withdrawButton = document.getElementById('withdraw-button');
+  var experienceIdElement = document.getElementById('experience-id');
+
+  if (withdrawButton && experienceIdElement) {
+    var experienceId = experienceIdElement.value;
+
+    withdrawButton.addEventListener('click', function(event) {
+      event.preventDefault();
+
+      if (confirm('Are you sure you want to withdraw the proposal?')) {
+        var xhr = new XMLHttpRequest();
+        xhr.open('POST', `/profile/${document.getElementById('username').value}/withdrawSummerExperience`, true);
+        xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+
+        xhr.onload = function() {
+          if (xhr.status === 200) {
+            console.log('Proposal withdrawn successfully');
+            window.location.href = `/cceMinor/${document.getElementById('username').value}/viewProposal`;
+          } else {
+            console.log('Error withdrawing proposal');
+          }
+        };
+
+        xhr.onerror = function() {
+          console.log('Request failed');
+        };
+
+        xhr.send('experience_id=' + encodeURIComponent(experienceId));
+      } else {
+        console.log('User cancelled withdrawal');
+      }
+    });
```

Comment:
We should be using jquery's ajax methods, like we do everywhere else

Reply:
Changed